src/analysis/schwartz_space.lean

```diff
@@ -102,6 +103,45 @@ lemma smooth (f : 𝓢(E, F)) (n : ℕ∞) : cont_diff ℝ n f := f.smooth'.of_l
 
 @[ext] lemma ext {f g : 𝓢(E, F)} (h : ∀ x, (f : E → F) x = g x) : f = g := fun_like.ext f g h
 
+section is_O
+
+variables (f : 𝓢(E, F))
+
+/-- Auxiliary lemma, used in proving the more general result `is_O_cocompact_zpow`. -/
+lemma is_O_cocompact_zpow_neg_nat (k : ℕ) :
```

Comment:
It won't work at `x = 0` (because `0 ^ (-k) = 0` but `f 0` might not vanish). So the statement is only true for filters `l` satisfying `∀ᶠ x in l, x ≠ 0`. I think the only reason to have this lemma at all is as a stepping stone to `is_O_cocompact_rpow` -- anyone who wants a concrete bound valid on the entire space can get one straight out of `f.decay`. (If you prefer, I can inline this lemma as a `have` statement inside the proof of the next lemma, to avoid cluttering the library and docs; but that will mean slower compilation time, cf. our discussion about `async` elsewhere in this PR.)